### PR TITLE
chore: add iterator timeout

### PR DESCRIPTION
**Pull request by weboko**
## Problem

Because iterator is using `while true` inside it is not getting terminated by `mocha` timeouts in local and sometime in pipeline environment.
It also is capable of blocking the main thread.

## Solution

Add options to configure timeout.
Add idle state to the generator.

### CODE REVIEW

File: packages/utils/src/common/to_async_iterator.ts [hunk 1]
```diff
@@ -23,8 +28,18 @@ export async function toAsyncIterator<T extends IDecodedMessage>(
...
+  const timeoutMs = iteratorOptions?.timeoutMs ?? 0;
+  const startTime = Date.now();
+
   async function* iterator(): AsyncIterator<T> {
     while (true) {
+      if (isWithTimeout && Date.now() - startTime >= timeoutMs) {
+        return;
+      }
+
+      await wait(60);
```
Comment (weboko) [inline packages/utils/src/common/to_async_iterator.ts:41]: This is required to make generator idle and unblock main thread.

Comment (fryorcraken) [reply packages/utils/src/common/to_async_iterator.ts:41]: Shall we add that as a code comment?

File: packages/utils/src/common/to_async_iterator.ts [hunk 2]
```diff
@@ -23,8 +28,18 @@ export async function toAsyncIterator<T extends IDecodedMessage>(
     options
   );
 
+  const isWithTimeout = Number.isInteger(iteratorOptions?.timeoutMs);
```
Comment (fryorcraken) [inline packages/utils/src/common/to_async_iterator.ts:31]: if timeout value is not specified, what happens in the browser?

Comment (weboko) [reply packages/utils/src/common/to_async_iterator.ts:31]: No timeout will be applied and execution will wait till resolved.

File: packages/utils/src/common/to_async_iterator.ts (hunk 1 above)
Comment (weboko) [reply packages/utils/src/common/to_async_iterator.ts:41]: Will add a js-doc to the function.

**Review by fryorcraken — APPROVED** [review]

**Pull request merged**